File: salimmo-parent/property-service/src/main/java/org/propertyservice/entities/InnerProperty.java
package org.propertyservice.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class InnerProperty extends BaseEntity implements Serializable {
    @OneToOne(mappedBy = "innerProperty")
    private Property property;
    private int nbrOfLevels;
    private int nbrOfPieces;
    private int nbrOfRooms;
    private int nbrOfRoomsGroundFloor;
    private int nbrOfParentalRooms;
    private int nbrOfShowerRooms;
    private int nbrOfShowerRoomsGroundFloor;
    private int nbrOfBathrooms;
    private int nbrOfBathroomsGroundFloor;
    private int nbrOfToilets;
    private int nbrOfToiletsGroundFloor;
    private String kitchenType;
    private Boolean existingBasement;
    private Boolean existingWashRoom;
    private Boolean existingCave;
    private int nbrOfCaves;
    private Boolean existingCellar;
}

File: salimmo-frontend/src/app/models/innerproperty/innerproperty.ts
import {Property} from "../property/property";

export class Innerproperty {
  id!:number;
  property!:Property;
  nbrOfLevels!:number;
  nbrOfPieces!:number;
  nbrOfRooms!:number;
  nbrOfRoomsGroundFloor!:number;
  nbrOfParentalRooms!:number;
  nbrOfShowerRooms!:number;
  nbrOfShowerRoomsGroundFloor!:number;
  nbrOfBathrooms!:number;
  nbrOfBathroomsGroundFloor!:number;
  nbrOfToilets!:number;
  nbrOfToiletsGroundFloor!:number;
  kitchenType!:string;
  existingBasement!:boolean;
  existingWashRoom!:boolean;
  existingCave!:boolean;
  nbrOfCaves!:number;
  existingCellar!:boolean;
}
